src/libraries/System.Security.Cryptography/src/System/Security/Cryptography/CngKey.StandardProperties.cs

```diff
@@ -279,10 +296,19 @@ public IntPtr ParentWindowHandle
         {
             get
             {
-                string? provider = _providerHandle.GetPropertyAsString(ProviderPropertyName.Name, CngPropertyOptions.None);
-                if (provider == null)
-                    return null;
-                return new CngProvider(provider);
+                if (!_hasCachedProvider || _keyHandle.IsClosed)
```

Comment:
```suggestion
                if (!_hasCachedProvider || _providerHandle.IsClosed)
```

The closed handle check is not using the same handle that's about to be queried.

Reply:
It doesn't matter in this particular case, but, agree it looks weird, so fixed.